### PR TITLE
Add docs for Merge operator

### PR DESCRIPTION
**Pull request by utkarsharma2**
# Description
## What is the current behavior?
In the past, we had a tutorial which illustrated how to use each of our operators/decorators:
https://github.com/astronomer/astro-sdk/blob/be6280df00ccff0d7a1c0dfb099b2065303dbe88/REFERENCE.md

closes: #591 
closes https://github.com/astronomer/astro-sdk/issues/530

## What is the new behavior?
Have a reference page per operator/decorator similar to
https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/operators/ecs.html#howto-operator-ecsoperator

In which we reference parts of an (automated tested) example DAG which illustrates the usage of that operator/decorator.

Many of these use cases already exist in our example DAGs - we should reference them.

merge
    with conflicts (what options the user has)
    without conflicts

## Does this introduce a breaking change?
Nope

### Checklist
- [X] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by utkarsharma2** [comment]
Addressed all the PR comments in - https://github.com/astronomer/astro-sdk/pull/629

File: docs/astro/sql/operators/load_file.rst [hunk 1]
```diff
@@ -40,7 +42,7 @@ Parameters to use when loading a file to the database table
 
     Note - When we are using ``if_exists='replace'`` we are dropping the existing table and then creating a new table. Here we are not reusing the schema.
 
-#. **output_table** - We can specify the output table to be created by passing in this parameter, which is expected to be an instance of ``astro.sql.table.Table``. Users can specify the schema of tables by passing in the ``columns`` parameter of ``astro.sql.table.Table`` object, which is expected to be a list of the instance of ``sqlalchemy.Column``. If the user doesn't specify the schema, the schema is inferred using pandas.
+#. **output_table** - We can specify the output table to be created by passing in this parameter, which is expected to be an instance of ``astro.sql.table.Table``. Users can specify the schema of tables by passing in the ``columns`` parameter of ``astro.sql.table.Table`` object, which is expected to be a list of the instance of `sqlalchemy.Column <https://docs.sqlalchemy.org/en/14/core/metadata.html#sqlalchemy.schema.Column>`_. . If the user doesn't specify the schema, the schema is inferred using pandas.
```
Comment (kaxil) [inline docs/astro/sql/operators/load_file.rst:45]: just an fyi - you can use intersphinx to reference classes or other objects from docs in other projects

Check https://docs.readthedocs.io/en/stable/guides/intersphinx.html

**Review by kaxil — APPROVED** [review]
Great work 👏

File: example_dags/example_merge.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+dag = DAG(
+    dag_id="example_merge_bigquery",
+    start_date=datetime(2019, 1, 1),
+    max_active_runs=3,
+    schedule_interval=timedelta(minutes=30),
+    default_args=default_args,
+)
+
+
```
Comment (phanikumv) [inline example_dags/example_merge.py:29]: Don't we need to showcase merge example for all the databases that we support, and usage of Constraints?

**Pull request merged**